api-event-files/index.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+$url = 'https://plausible.io/api/event';
+
+$headers = [];
+
+$allowed_headers =
+
+	[
+		'Accept',
+		'Accept-Language',
+		'Connection',
+		'Content-Length',
+		'Content-Type',
+		'Cookie',
+		'Origin',
+		'Referer',
+		'Sec-Ch-Ua',
+		'Sec-Ch-Ua-Mobile',
+		'Sec-Ch-Ua-Platform',
+		'Sec-Fetch-Dest',
+		'Sec-Fetch-Mode',
+		'Sec-Fetch-Site',
+		'User-Agent',
+		'X-Forwarded-Host',
+		'X-Forwarded-Port',
+		'X-Forwarded-Proto',
+		'X-Forwarded-Scheme',
+		'X-Original-Host',
+		'X-Real-Ip',
+		'X-Request-Id',
+		'X-Scheme',
+	];
+
+foreach ( getallheaders() as $key => $value ) {
+	if ( in_array( $key, $allowed_headers ) ) {
+		$headers[ $key ] = sprintf( '%s: %s', $key, $value );
+	}
+}
+
+
+if ( ! empty( $_SERVER['HTTP_CLIENT_IP'] ) ) {
+	$ip = $_SERVER['HTTP_CLIENT_IP'];
+} elseif ( ! empty( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+	$ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
+} else {
+	$ip = $_SERVER['REMOTE_ADDR'];
+}
+
+$headers['X-Forwarded-For'] = 'X-Forwarded-For: ' . $headers['X-Real-Ip'];
```

Comment:
What does this line do? If I'm reading it right it overwrites the `X-Forwarded-For` header with the content of `X-Real-Ip`, is that correct?

Reply:
To get accurate data from the viewer, we need to pass the real viewer API not the server one.
I fixed this by using the $ip var above, will update the code.

Reply:
> To get accurate data from the viewer, we need to pass the real viewer API not the server one.

Yeah exactly. To get the visitor address there are 2 scenarios:
1. No proxy in front of WP: take the `remote_addr` and send it as `X-Forwarded-For` to plausible.io
2. Proxy in front of WP: take the value of `X-Forwarded-For`, add the current server IP to the end of the list, and forward it to plausible.io

On our side we parse `X-Forwarded-For` according to the spec:
> If a request goes through multiple proxies, the IP addresses of each successive proxy is listed. This means that, given well-behaved client and proxies, the rightmost IP address is the IP address of the most recent proxy and the leftmost IP address is the IP address of the originating client.

Reply:
I'm still confused about the purpose of the `$ip` variable above and how exactly the visitor IP is resolved. At the moment it looks like we're taking the `X-Real-Ip` header value, is that correct? Who/What sets that header in the first place?